### PR TITLE
[FIX] calendar: allow admin to edit any event

### PR DESCRIPTION
**Pull request by nboulif**
Steps to reproduce:

  - Install `calendar` module
  - With demo user, go to calendar and create a new event
  - With admin user, go to the calendar and try to edit the event

Issue:

  Admin can't edit the event.

Cause:

  Only allowing the user who created the event or who is in the
  attendees list to edit it.

Solution:

  Allow to edit the event if user is admin (or created the event
  or is in attendees list).

opw-3185743

### CODE REVIEW

File: addons/calendar/static/src/views/attendee_calendar/common/attendee_calendar_common_popover.js [hunk 1]
```diff
@@ -62,18 +62,18 @@ export class AttendeeCalendarCommonPopover extends CalendarCommonPopover {
...
-        return super.isEventDeletable && this.isCurrentUserAttendee && !this.isEventArchivable;
+        return super.isEventDeletable && this.isEventEditable && !this.isEventArchivable;
     }
 
     /**
      * @override
      */
     get isEventEditable() {
-        return this.isCurrentUserAttendee;
+        return this.props.record.rawRecord.user_can_edit;
```
Comment (nle-odoo) [inline addons/calendar/static/src/views/attendee_calendar/common/attendee_calendar_common_popover.js:72]: the code should work without user_can_edit in the view, the majority of installations will not update the calendar module and so the field will not be available for them

so I guess this should fallback on `this.isCurrentUserAttendee`

**Review by nle-odoo — APPROVED** [review]
looks ok if it's what we want functionally

File: addons/calendar/static/src/views/attendee_calendar/common/attendee_calendar_common_popover.js (hunk 1 above)
Comment (nboulif) [reply addons/calendar/static/src/views/attendee_calendar/common/attendee_calendar_common_popover.js]: Indeed, updated :+1:

**Comment by nboulif** [comment]
@nle-odoo Thanks for the review!
:warning: Remove in `Master` backward compatibility for `user_can_edit` field in JS files (because missing in view in stable without upgrade).

**Comment by nboulif** [comment]
ping @odoo/rd-subscription-sign-calendar

**Comment by arj-odoo** [comment]
@robodoo r+

**Pull request closed without merging**